std-bits/base/src/main/java/org/enso/base/cache/LRUCache.java

```diff
@@ -344,6 +355,11 @@ public LRUCacheSettings getSettings() {
     return settings;
   }
 
+  /** Public for testing. */
+  public void simulateReloadTestOnly() {
+    reloadDetector.simulateReloadTestOnly();
+  }
```

Comment:
I'm not convinced about this mock.

After all - tests relying on this mock use _completely different_ logic for checking if the cache should be cleared than what will be running in the IDE. The `Reload_Detector` is a new component and it is not tested anywhere at all.

I think it would be best to add a test to the JVM tests, similar to something like `RuntimeExecutionEnvironmentTest` that checks the refresh logic is correctly connected through the Language Server. Perhaps we don't need to test it all the way down to HTTP. It would probably suffice to have a test that instantiates the Java `ReloadDetector` and checks that it indeed detects cache invalidation correctly.

Reply:
> add a test to the JVM tests, similar to something like `RuntimeExecutionEnvironmentTest` that checks the refresh logic is correctly connected through the Language Server.

Logic of `Managed_Resource.register ref on_finalize Boolean.True` is **unit tested** at 

https://github.com/enso-org/enso/blob/cc0b020e1eb2391f1af21e4a3c3edb4fb720b6c2/engine/runtime-integration-tests/src/test/java/org/enso/interpreter/runtime/ManagedResourceTest.java#L96

If we have a unit test of `Managed_Resource` and another unit test for `ReloadDetector`, then I believe we have [100% coverage](http://wiki.apidesign.org/wiki/CodeCoverage)!

Reply:
My point was that we are not testing the _integration_ between `ReloadDetector` and `Managed_Resource` logics if we rely on a mock when testing `ReloadDetector`.

The problem is clear in the `get` bug which if I understand correctly you agree with me is a bug - we are not catching the possible `Uninitialized_State` there, possibly returning dataflow error instead of boolean. This is not covered by any existing test as far as I can tell. Given that we seem to have a bug and no test seems to be catching the bug - I'd say we don't have good enough tests for this facility.

Reply:
I agree that a test like this is necessary -- precisely because I have already had the tests pass but the feature fail to work. Thus: #11792.